Cabal/Distribution/Simple/Test/LibV09.hs

```diff
@@ -212,7 +213,12 @@ stubMain :: IO [Test] -> IO ()
 stubMain tests = do
     (f, n) <- fmap read getContents
     dir <- getCurrentDirectory
-    results <- tests >>= stubRunTests
+    let errhandler :: SomeException -> IO TestLogs
+        errhandler e = do
+          return $ TestLog { testName = "Cabal test suite exception",
+                             testOptionsReturned = [],
+                             testResult = Error $ displayException e }
```

Comment:
As a matter of style, can I suggest that you move this handler to a `where` clause, since it doesn't need to capture any earlier context and it reads better to gets the exception case out of the main flow.

Otherwise, it looks fine (modulo compile errors with older library versions).